### PR TITLE
Styling

### PR DESCRIPTION
**Pull request by josie-garza**
- added bar chart for numerical and time stamp columns detail panels
- added pie chart for categorical column detail panels
- updated backend calls for column specifications
- added tests for the updated backend

### CODE REVIEW

File: jupyterlab_automl/jupyterlab_automl/handlers.py [hunk 1]
```diff
@@ -60,109 +95,141 @@ def create_automl_parent(client):
...
+        return "[" + str(bucket.min) + ", " + str(round(bucket.max)) + "]"
+    elif bucket.max == float("inf"):
+        return "[" + str(round(bucket.min)) + ", " + str(bucket.max) + "]"
+    else:
+        return "[" + str(round(bucket.min)) + ", " + str(round(bucket.max)) + "]"
+
+
+def get_detail_panel(column_spec, count):
+    chart_data = []
+    if column_spec.data_type.type_code == 3:
```
Comment (shunr) [inline jupyterlab_automl/jupyterlab_automl/handlers.py:109]: Can we use enums here? I feel like having hardcoded integers everywhere could make things harder ro maintain in the future

File: jupyterlab_automl/jupyterlab_automl/handlers.py [hunk 2]
```diff
@@ -60,109 +95,141 @@ def create_automl_parent(client):
...
             {
                 "id": column_spec.name,
                 "dataType": type_code,
                 "displayName": column_spec.display_name,
                 "distinctValueCount": column_spec.data_stats.distinct_value_count,
-                "invalidValueCount": rowCount - column_spec.data_stats.valid_value_count,
-                "nullValueCount": str(column_spec.data_stats.null_value_count) + " (" + str(math.floor(column_spec.data_stats.null_value_count / rowCount * 100)) + "%)",
+                "invalidValueCount": table_spec.row_count
+                - column_spec.data_stats.valid_value_count,
+                "nullValueCount": str(column_spec.data_stats.null_value_count)
```
Comment (shunr) [inline jupyterlab_automl/jupyterlab_automl/handlers.py:160]: why not have this in the Typescript code? the api returned a preformatted string could be confusing if we think it should return a number

File: jupyterlab_automl/jupyterlab_automl/handlers.py [hunk 3]
```diff
@@ -60,109 +95,141 @@ def create_automl_parent(client):
...
-                "columnSpecs": column_specs,
-                "chartSummary": chart_summary,
+                "columnSpecs": get_column_specs(client, table_spec)[0],
+                "chartSummary": get_column_specs(client, table_spec)[1],
             }
-        ) 
-    return {
-        "tableSpecs": table_specs
-    }
+        )
```
Comment (shunr) [inline jupyterlab_automl/jupyterlab_automl/handlers.py:192]:
```
columSpecs, chartSummary = get_column_specs(client, table_spec)
...
{
"columnSpecs": columnSpecs,
"chartSummary": chartSummary
}
```
I think we should try to call to get_column_specs only once

File: jupyterlab_automl/jupyterlab_automl/handlers.py [hunk 4]
```diff
@@ -60,109 +95,141 @@ def create_automl_parent(client):
...
+    return {"tableSpecs": table_specs}
+
+
+def get_dataset_metadata(dataset):
+    if dataset.name.split("/")[-1][:3] == "TBL":
```
Comment (shunr) [inline jupyterlab_automl/jupyterlab_automl/handlers.py:197]: maybe abstract out a helper function like
```
def parse_dataset_type(id: string):
    return dataset.name.split("/")[-1][:3]
```

File: jupyterlab_automl/src/components/list_resources_table.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
+import { CSSProperties } from '@material-ui/core/styles/withStyles';
```
Comment (shunr) [inline jupyterlab_automl/src/components/list_resources_table.tsx:1]: Probably can delete this file cause it was moved to the shared folder

File: jupyterlab_automl/src/components/dataset_component.tsx [hunk 6]
```diff
@@ -0,0 +1,493 @@
+import { Box, Grid, LinearProgress } from '@material-ui/core';
+import * as csstips from 'csstips';
+import * as React from 'react';
+import { stylesheet } from 'typestyle';
+import MaterialTable from 'material-table';
+import { DatasetService, TableSpec, Dataset } from '../service/dataset';
+import { style, ColumnType } from './list_resources_table';
```
Comment (shunr) [inline jupyterlab_automl/src/components/dataset_component.tsx:7]:
```suggestion
import { style, ColumnType } from './shared/list_resources_table';
```

File: jupyterlab_automl/jupyterlab_automl/handlers.py (hunk 1 above)
Comment (josie-garza) [reply jupyterlab_automl/jupyterlab_automl/handlers.py:109]: Agreed. Perhaps we should also have an enumeration for dataset type?

File: jupyterlab_automl/jupyterlab_automl/handlers.py (hunk 3 above)
Comment (josie-garza) [reply jupyterlab_automl/jupyterlab_automl/handlers.py:192]: nice catch i thought i had removed this...

File: jupyterlab_automl/jupyterlab_automl/handlers.py (hunk 4 above)
Comment (josie-garza) [reply jupyterlab_automl/jupyterlab_automl/handlers.py:197]: good idea

**Comment by josie-garza** [comment]
- add dataset types for image object
- filter only datasets we support
- disable launching model and image datasets

**Pull request merged**